dev/pytest

```diff
@@ -38,17 +38,19 @@ if [ -n "$KOALAS_USAGE_LOGGER" ]; then
 fi
 
 # Runs both doctests and unit tests by default, otherwise hands arguments over to pytest.
+# we're doing pytest with `xvfb-run` since without this some of the tests would fail for GUI library because there is no display server.
+# The X virtual framebuffer (Xvfb) display server is used to perform all the graphical operations in memory without showing any screen output.
 if [ "$#" = 0 ]; then
     if [[ "$SPARK_VERSION" == 2.3* ]] || [[ "$SPARK_VERSION" == 2.4.1* ]] || [[ "$SPARK_VERSION" == 2.4.2* ]]; then
         # Delta requires Spark 2.4.2+. We skip the related doctests.
         if [[ "$SPARK_VERSION" == 2.3* ]]; then
-            pytest --cov=databricks -k " -melt -to_delta -read_delta" --verbose --showlocals --doctest-modules databricks "${logopts[@]}"
+            xvfb-run python -m pytest --cov=databricks -k " -melt -to_delta -read_delta" --verbose --showlocals --doctest-modules databricks "${logopts[@]}"
```

Comment:
@itholic, in this way, it requires `xvfb-run` to the dev env even where the display is available. Can you factor this out as an environment variable and set it in Github Actions only?

e.g.):

In this file:

```bash
PYTHON_EXECUTABLE="${PYTHON_EXECUTABLE:-python}"
...
"$PYTHON_EXECUTABLE" -m pytest --cov=databricks -k ...
```

In Github Actions, where it's required,

```bash
PYTHON_EXECUTABLE="xvfb-run python" ./dev/pytest ...
```

Reply:
that makes sense. thanks for the comment!